src/DateInput.tsx

```diff
@@ -423,7 +423,7 @@ export default function DateInput({
      * However, given 2, smallers possible number would be 20, and thus keeping the focus in
      * this field doesn't make sense.
      */
-    if (Number(value) * 10 > Number(max) || value.length >= max.length) {
+    if (Number(value) && (Number(value) * 10 > Number(max) || value.length >= max.length)) {
```

Comment:
actual issue here let say ur max month is 4 and user enter 0 
value.length >= max.length will for the user to next input